data/soapvalidator/Briefcase/Briefcase_DocumentAction.xml

```diff
@@ -0,0 +1,355 @@
+                          <doc l="${account1.folder.briefcase.id}">
+                            <upload id="${document.pdffile.aid}"/>
+                          </doc>
+                        </SaveDocumentRequest>
+        </t:request>
+        <t:response>
+            <t:select path="//mail:SaveDocumentResponse/mail:doc" attr="id" set="document.pdffile.id"/>
+		
+        </t:response>
+    </t:test>
+	
+    <t:test>
+        <t:request>
+                        <DocumentActionRequest xmlns="urn:zimbraMail">
+                            <action id="${document.pdffile.id}" op="watch"/>
+                        </DocumentActionRequest>
+        </t:request>
+        <t:response>
+            <t:select path="//mail:DocumentActionResponse/mail:action" attr="id" match="${document.pdffile.id}" />
+            <t:select path="//mail:DocumentActionResponse/mail:action" attr="op" match="watch"/>
+        </t:response>
+    </t:test>
+
+    <t:test>
+        <t:request>
+                        <GetWatchingItemsRequest xmlns="urn:zimbraMail">
+                        </GetWatchingItemsRequest>
+        </t:request>
+        <t:response>
+            <t:select path="//mail:GetWatchingItemsResponse/mail:target/mail:item" attr="id" match="${document.pdffile.id}"/>
+            <t:select path="//mail:GetWatchingItemsResponse/mail:target" attr="email" match="${account1.name}"/>
+        </t:response>
+    </t:test>
+
+    <t:test>
+        <t:request>
+                        <GetNotificationsRequest xmlns="urn:zimbraMail">
+                        </GetNotificationsRequest>
+        </t:request>
+        <t:response>
+            <t:select path="//mail:GetNotificationsResponse/mail:op" attr="name" match="Watch"/>
+            <t:select path="//mail:GetNotificationsResponse/mail:a" attr="email" match="${account1.name}"/>
+            <t:select path="//mail:GetNotificationsResponse/mail:a" attr="itemId" match="${document.pdffile.id}"/>
+        </t:response>
+    </t:test>
+
+    <t:test>
+        <t:request>
+                        <DocumentActionRequest xmlns="urn:zimbraMail">
+                            <action id="${document.pdffile.id}" op="!watch"/>
+                        </DocumentActionRequest>
+        </t:request>
+        <t:response>
+            <t:select path="//mail:DocumentActionResponse" attr="op" match="!watch"/>" />
+        </t:response>
+    </t:test>
+
+    <t:test>
+        <t:request>
+                        <GetWatchingItemsRequest xmlns="urn:zimbraMail">
+                        </GetWatchingItemsRequest>
+        </t:request>
+        <t:response>
+            <t:select path="//mail:GetWatchingItemsResponse"/>
+        </t:response>
+    </t:test>
+
+    <t:test>
+        <t:request>
+                        <GetNotificationsRequest xmlns="urn:zimbraMail">
+                        </GetNotificationsRequest>
+        </t:request>
+        <t:response>
+            <t:select path="//mail:GetNotificationsResponse"/>
```

Comment:
Add validation to check item is not watched (op name="Unwatch")

Reply:
Same is the case with GetNotificationResponse.
```
  <soap:Body>
    <GetNotificationsResponse lastSeen="0" xmlns="urn:zimbraMail"/>
  </soap:Body>
```

Reply:
Same as above - verify absence of element

Reply:
Added required validation.